### PR TITLE
#19: Install TDD Framework

### PR DESCRIPTION
**Pull request by benniemosher**
Alright, so I decided to break the whole testing thing out into multiple smaller branches so that the PRs do not get so daunting. This is the first of probably many.

This PR simply sets up the TDD framework that we should need to begin writing the tests for the plugin. Here is a small recap of what has been installed:

As advised from here: http://karma-runner.github.io/0.13/intro/installation.html

- karma
- karma-jasmine
- karma-chrome-launcher
- jasmine-core (There appears to be some bug in npm that was not installing dependencies of modules, so I had to manually install this.)

As advised from here: https://github.com/9joneg/karma-sinon-chrome

- karma-sinon-chrome

This so far should be all that we need. I have created a small test spec just to make sure that all of the tools are working.

Basically to install you just need to run `npm install` and it should install all of the NPM modules that you need. Then from what I can tell (I am new to Karma), you run `karma start` that opens a chrome browser and will run your tests every time you save a change.

If you want to quickly run the full suite, instead of just have it watch files, in another terminal window/tab just type `karma run` while `karma start` is still running and it will run your specs for you.

Once I get your approval on this PR and you are able to run the test yourself, I will start with the first actual spec and submit PRs for each separate spec file.

Let me know if you have any questions!

### CODE REVIEW

**Comment by benniemosher** [comment]
One error that I am getting right now is pasted below. I am not sure if this is because I haven't used sinon-chrome to initialize or install the plugin yet. If you don't mind take a quick look at it, I would appreciate it. Chrome plugins are new to me, so I am still learning the ropes.

```
USS-Kelvin [devtools-author] (issue-#19/install-sinon-chrome) $ karma start
09 02 2016 16:23:39.140:WARN [karma]: No captured browser, open http://localhost:9876/
09 02 2016 16:23:39.157:INFO [karma]: Karma v0.13.19 server started at http://localhost:9876/
09 02 2016 16:23:39.167:INFO [launcher]: Starting browser Chrome
09 02 2016 16:23:41.007:INFO [Chrome 48.0.2564 (Mac OS X 10.11.3)]: Connected on socket /#VLUutoLJWC3ulCwKAAAA with id 78999512
Chrome 48.0.2564 (Mac OS X 10.11.3) ERROR
  Uncaught TypeError: Cannot read property 'panels' of undefined
  at /Users/benniemosher/Engineering/devtools-author/app/scripts/devtools.js:9

09 02 2016 16:23:41.411:WARN [web-server]: 404: /dist/scripts/themes.json
```

**Comment by micjamking** [comment]
@benniemosher Awesome! You are on the man, I really appreciate this. :+1: 

**re: `Uncaught TypeError: Cannot read property 'panels' of undefined` Error**
After digging around a bit, it looks like the error is due to the version of `sinon-chrome` (v0.2.1) the `karma-sinon-chrome` plugin is [dependent on](https://github.com/9joneg/karma-sinon-chrome/blob/master/package.json#L8) does not support `chrome.devtools.*` API; this was not added until `sinon-chrome` [v1.0.2](https://github.com/acvetkov/sinon-chrome/releases/tag/1.0.2). 

I tried updating the dependency within the `karma-sinon-chrome` plugin, but the latest version of `sinon-chrome` uses ES6 modules and subsequently requires webpack, so I'm not sure how to get everything hooked up to run the latest version of `sinon-chrome` with `karma-sinon-chrome`. At the moment, I'm not sure how to include `sinon-chrome` without the `karma-sinon-chrome` plugin.

I'm fine with merging this in for now so you can continue progressing, however hopefully we can find a way to integrate with the latest version of `sinon-chrome` so we have access to the `chrome.devtools.*` API

**Comment by micjamking** [comment]
Reference: Issue #19
